internal/service/elbv2/hosted_zone_id_data_source.go

```diff
@@ -0,0 +1,117 @@
+	endpoints.EuNorth1RegionID:     "Z23TAZ6LKFMNIO",
+	endpoints.EuSouth1RegionID:     "Z3ULH7SSC9OV64",
+	endpoints.EuWest1RegionID:      "Z32O12XQLNTSW2",
+	endpoints.EuWest2RegionID:      "ZHURV8PSTC4K8",
+	endpoints.EuWest3RegionID:      "Z3Q77PNBQS71R4",
+	endpoints.MeSouth1RegionID:     "ZS929ML54UICD",
+	endpoints.SaEast1RegionID:      "Z2P70J7HTTTPLU",
+	endpoints.UsEast1RegionID:      "Z35SXDOTRQ7X7K",
+	endpoints.UsEast2RegionID:      "Z3AADJGX6KTTL2",
+	endpoints.UsGovEast1RegionID:   "Z166TLBEWOO7G0",
+	endpoints.UsGovWest1RegionID:   "Z33AYJ8TM3BH4J",
+	endpoints.UsWest1RegionID:      "Z368ELLRRE2KJ0",
+	endpoints.UsWest2RegionID:      "Z1H1FL5HABSF5",
+}
+
+// See https://docs.aws.amazon.com/general/latest/gr/elb.html#elb_region
+var HostedZoneIdPerRegionNLBMap = map[string]string{
+	endpoints.AfSouth1RegionID:     "Z203XCE67M25HM",
+	endpoints.ApEast1RegionID:      "Z12Y7K3UBGUAD1",
+	endpoints.ApNortheast1RegionID: "Z31USIVHYNEOWT",
+	endpoints.ApNortheast2RegionID: "ZIBE1TIR4HY56",
+	endpoints.ApNortheast3RegionID: "Z1GWIQ4HH19I5X",
+	endpoints.ApSouth1RegionID:     "ZVDDRBQ08TROA",
+	endpoints.ApSoutheast1RegionID: "ZKVM4W9LS7TM",
+	endpoints.ApSoutheast2RegionID: "ZCT6FZBF4DROD",
+	endpoints.ApSoutheast3RegionID: "Z01971771FYVNCOVWJU1G",
+	endpoints.CaCentral1RegionID:   "Z2EPGBW3API2WT",
+	endpoints.CnNorth1RegionID:     "Z3QFB96KMJ7ED6",
+	endpoints.CnNorthwest1RegionID: "ZQEIKTCZ8352D",
+	endpoints.EuCentral1RegionID:   "Z3F0SRJ5LGBH90",
+	endpoints.EuNorth1RegionID:     "Z1UDT6IFJ4EJM",
+	endpoints.EuSouth1RegionID:     "Z23146JA1KNAFP",
+	endpoints.EuWest1RegionID:      "Z2IFOLAFXWLO4F",
+	endpoints.EuWest2RegionID:      "ZD4D7Y8KGAS4G",
+	endpoints.EuWest3RegionID:      "Z1CMS0P5QUZ6D5",
+	endpoints.MeSouth1RegionID:     "Z3QSRYVP46NYYV",
+	endpoints.SaEast1RegionID:      "ZTK26PT1VY4CU",
+	endpoints.UsEast1RegionID:      "Z26RNL4JYFTOTI",
+	endpoints.UsEast2RegionID:      "ZLMOA37VPKANP",
+	endpoints.UsGovEast1RegionID:   "Z1ZSMQQ6Q24QQ8",
+	endpoints.UsGovWest1RegionID:   "ZMG1MZ2THAWF1",
+	endpoints.UsWest1RegionID:      "Z24FKFUX50B4VW",
+	endpoints.UsWest2RegionID:      "Z18D5FSROUN65G",
+}
+
+func DataSourceHostedZoneID() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceHostedZoneIDRead,
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"load_balancer_type": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Default:      elbv2.LoadBalancerTypeEnumApplication,
+				ValidateFunc: validation.StringInSlice([]string{"application", "network"}, false), // should only accept application or network
+			},
+		},
+	}
+}
+
+func dataSourceHostedZoneIDRead(d *schema.ResourceData, meta interface{}) error {
+	region := meta.(*conns.AWSClient).Region
+	if v, ok := d.GetOk("region"); ok {
+		region = v.(string)
+	}
+
+	var lbType string
+	lbType = elbv2.LoadBalancerTypeEnumApplication
```

Comment:
Good use of the in-build values supplied by the SDK. A further addition would be to simplify this to the following:

```suggestion
	lbType := elbv2.LoadBalancerTypeEnumApplication

Reply:
good point, I incorporated it in [a2e453e](https://github.com/hashicorp/terraform-provider-aws/pull/24749/commits/a2e453e420e2a9a17a8029ca4e45aca5b4b83911)